lib/Geo/GeoJson.cpp

```diff
@@ -36,19 +36,83 @@
 #include <s2/s2point_region.h>
 #include <s2/s2polygon.h>
 #include <s2/s2polyline.h>
+#include <s2/util/coding/coder.h>
 
 #include "Basics/StringUtils.h"
 #include "Basics/VelocyPackHelper.h"
 #include "Basics/debugging.h"
 #include "Geo/GeoParams.h"
 #include "Geo/ShapeContainer.h"
 #include "Geo/S2/S2MultiPointRegion.h"
-#include "Geo/S2/S2MultiPolyline.h"
+#include "Geo/S2/S2MultiPolylineRegion.h"
 #include "Logger/Logger.h"
 
 namespace arangodb::geo::json {
 namespace {
 
+template<bool Validation>
+S2Point encodePointImpl(S2LatLng latLng, coding::Options options,
+                        Encoder* encoder) noexcept {
+  if constexpr (Validation) {
```

Comment:
assert that encoder is present only if Validation is set to true? To make sure we accidently do not skip encoding.

Reply:
Done